firebase-firestore/src/main/java/com/google/firebase/firestore/local/Persistence.java

```diff
@@ -83,7 +83,7 @@
    * implementation to the extent possible (e.g. in the case of uid switching from
    * sally=>jack=>sally, sally's mutation queue will be preserved).
    */
-  abstract MutationQueue getMutationQueue(User user);
+  abstract MutationQueue getMutationQueue(User user, IndexManager indexManager);
```

Comment:
Note that this interaction is kind of weird since we also get the IndexManager form persistence. The other alternative is that we keep a user-scoped map of existing IndexManagers in the persistence implementations, but I think that is worse.